src/cargo/ops/cargo_compile.rs

```diff
@@ -163,6 +167,23 @@ pub fn compile_pkg(package: &Package, options: &CompileOptions)
     let to_build = packages.iter().find(|p| p.package_id() == pkgid).unwrap();
     let targets = try!(generate_targets(to_build, mode, filter, release));
 
+    let target_with_args = match target_rustc_args {
+        &Some(args) => {
```

Comment:
Typically instead of matching `&T` we use `match *target_rustc_args` to dereference sooner. You may also be able to compress this a bit via:

```rust
match *target_rustc_args {
    Some(args) if args.len() > 1 => return Err(...),
    Some(args) => { ... }
    // ...
}
```